src/components/organisms/MapBoxMap/MapboxMap.tsx

```diff
@@ -77,20 +78,33 @@ const MapboxMap: React.FC = () => {
         if (event.action === 'select-option') {
             if (event.option?.value) {
                 setTimestampToFetch(event.option.value)
+                setCurrentSelectedTimestamps([...currentSelectedTimestamps, event.option.value])
             }
         } else if (event.action === 'remove-value') {
             if (event.removedValue?.value && currentAoiId) {
                 removePredictionById(map!, event.removedValue.value, currentAoiId)
+                setCurrentSelectedTimestamps(currentSelectedTimestamps.filter((ts) => ts !== event.removedValue.value))
             }
         } else if (event.action === 'clear') {
             if (map) {
                 removeAllPredictions(map)
+                setCurrentSelectedTimestamps([])
             }
         }
 
         return
     }
 
+
+    function handleProbabilityFilterChange(e: React.ChangeEvent<HTMLInputElement>) {
+        for (let timestamp of currentSelectedTimestamps) {
+            const predictionLayerId = `prediction-${timestamp}-${currentAoiId}`
```

Comment:
this needs to change to:  const predictionLayerId = `prediction-${timestamp}-${currentAoiId}-point`
(I added a heatmap layer but to still have the distinct points I kept the existing point layer, we have now 2 preditcion layers. To distinguish them I added "-heatmap" and  "-point" to the layerId.